temporalio/common.py

```diff
@@ -962,6 +962,47 @@ def set(
 
 MetricMeter.noop = _NoopMetricMeter()
 
+
+@dataclass(frozen=True)
+class Priority:
+    """Priority contains metadata that controls relative ordering of task processing when tasks are
+    backlogged in a queue. Initially, Priority will be used in activity and workflow task queues,
+    which are typically where backlogs exist.
+
+    Priority is (for now) attached to workflows and activities. Activities and child workflows
+    inherit Priority from the workflow that created them, but may override fields when they are
+    started or modified. For each field of a Priority on an activity/workflow, not present or equal
+    to zero/empty string means to inherit the value from the calling workflow, or if there is no
+    calling workflow, then use the default (documented below).
+
+    The overall semantics of Priority are:
+    1. First, consider "priority_key": lower number goes first.
+    (more will be added here later)
+    """
+
+    priority_key: int = 0
```

Comment:
I don't think this should have a default at this time (but we can change that later if we'd like).

Also, even though proto and Go don't have the ability to model optional, we do, so we can treat zero as `Optional` here which may make sense. Of course we can still accept `0` or `None` to mean `None`, but when relaying to a user, I think `None` makes more sense.

Reply:
Sure, I can agree with that